packages/search/src/use-fuzzy-search.ts

```diff
@@ -30,33 +30,26 @@ export const useFuzzySearch = < T >( {
 	 * We want to re-use the `Fuse` instance because creating one every time
 	 * the dataset changes is an expensive operation.
 	 *
-	 * I first used `useRef` to save that instance, but that hook does not allow
-	 * lazy instantiation of the initial value, so I had to create wrapper functions
-	 * and the result didn't look great.
-	 *
-	 * Falling back to `useState` fixes that problem, though now we have to add
-	 * the `fuseInstance` variable to the dependency array.
-	 *
 	 */
-	const [ fuseInstance ] = useState( () => {
-		return new Fuse( data, {
+	const fuseInstance = useMemo( () => {
+		return new Fuse( [], {
 			keys,
 			includeScore: false,
 			includeMatches: false,
 			...options,
 		} );
-	} );
-
-	useEffect( () => {
-		fuseInstance.setCollection( data );
-	}, [ fuseInstance, data ] );
+	}, [ keys, options ] );
 
 	const results = useMemo( () => {
 		if ( ! query ) {
 			return data;
 		}
 
-		return fuseInstance.search( query ).map( ( { item } ) => item );
+		// Every time the query or the data changes, we update the collection
+		// This assignment takes less than 1ms for thousands of items
+		fuseInstance.setCollection( data );
+		const results = fuseInstance.search( query ).map( ( { item } ) => item );
+		return results;
```

Comment:
I don't think we need this `results` variable since we're immediately returning it:

```suggestion

		return fuseInstance.search( query ).map( ( { item } ) => item );
```

Would love a blank line to separate the `setCollection` statement from the `search` invocation, though that's nitpicking at this point hehe :)

Reply:
Done !! 😎 

Thanks for the feedback and the slack discussion 💪 . I appreciate it very much !!